### PR TITLE
Removes a (potentially) no-longer-needed fix from character previews

### PR DESCRIPTION
**Pull request by vinylspiders**
## About The Pull Request

EDIT: After speaking with folks in discord, the code that was causing this may not even be required anymore.

This is currently being tested to make sure it is in fact no longer needed.

edit edit: currently is TMed on Nova, so far everyone reporting the issue has been on version 1589. Bug is allegedly fixed in Byond 1609+.

edit edit edit: Just got someone on 1620 who has the tiny character preview bug. Looks like it's not actually fixed, at least on that version...

## Why It's Good For The Game

Bugfix

## Changelog

:cl:
code: removes a tg-fix for char previews not showing until hovered or rotated that has been fixed as of byond version 1609+
/:cl:

### CODE REVIEW

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -88,7 +88,11 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	/// If set to TRUE, will update character_profiles on the next ui_data tick.
 	var/tainted_character_profiles = FALSE
 
+	/// the timerid of the char_preview/updatebody() callback
+	var/character_preview_view_timerid
+
 /datum/preferences/Destroy(force)
+	deltimer(character_preview_view_timerid)
```
Comment (LemonInTheDark) [inline code/modules/client/preferences.dm:95]: there's no need for this, DELETE_ME handles it on its own

File: code/modules/client/preferences.dm [hunk 2]
```diff
@@ -88,7 +88,11 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	/// If set to TRUE, will update character_profiles on the next ui_data tick.
 	var/tainted_character_profiles = FALSE
 
+	/// the timerid of the char_preview/updatebody() callback
+	var/character_preview_view_timerid
```
Comment (LemonInTheDark) [inline code/modules/client/preferences.dm:92]: this name is bad and you're using the comment to do some of the heavy lifting
rather then naming the id after the object it is targeting, it seems more sensible to name it after the thing the timer does

File: code/modules/client/preferences.dm [hunk 3]
```diff
@@ -149,7 +153,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 		// Lemon from the future: this issue appears to replicate if the byond map (what we're relaying here)
 		// Is shown while the client's mouse is on the screen. As soon as their mouse enters the main map, it's properly scaled
 		// I hate this place
-		addtimer(CALLBACK(character_preview_view, TYPE_PROC_REF(/atom/movable/screen/map_view/char_preview, update_body)), 1 SECONDS)
+		character_preview_view_timerid = addtimer(CALLBACK(character_preview_view, TYPE_PROC_REF(/atom/movable/screen/map_view/char_preview, update_body)), 1 SECONDS, TIMER_STOPPABLE | TIMER_DELETE_ME)
```
Comment (ZephyrTFA) [inline code/modules/client/preferences.dm:156]: all you need here is the delete me flag, everything else can go

[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/modules/client/preferences.dm:156]: scratch that, we do already, just do https://github.com/tgstation/tgstation/pull/80867#discussion_r1445542191

File: code/modules/client/preferences.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/client/preferences.dm:95]: redundant but potentially useful to know

**Pull request closed without merging**